### ISSUE TITLE
Support for R

### ISSUE THREAD

**Issue by bryancross** [dependabot/dependabot-core#2042]
Support for R.  This comes up in every case where data scientists are considering GitHub.

**Comment by dfrankow** [comment]
I think the big ticket item is [CRAN](https://cran.r-project.org/). See https://r-pkgs.org/release.html: _"If you want your package to have significant traction in the R community, you need to submit it to CRAN."_

**Comment by dfrankow** [comment]
The most mature library I know of for dealing with R packages (including structuring, checking, and releasing) is [devtools](https://www.rdocumentation.org/packages/devtools/versions/2.3.2).

Packages are more generally described in https://r-pkgs.org/intro.html.

The file that contains dependencies to be checked is DESCRIPTION, see an example of the "Depends" block [here](https://github.com/r-lib/devtools/blob/master/DESCRIPTION).

**Comment by dfrankow** [comment]
See also more info at https://r-pkgs.org/description.html#dependencies.

**Comment by cderv** [comment]
See also **renv** R 📦 which allows to write a `renv.lock` file than contains the version number of the package, and the source if they come from CRAN or a source repo like Github. Any R project can use this to fix their dependencies requirement.

Otherwise, yes `DESCRIPTION` file are in every R packages developed to inform about dependencies and minimum version used. But it does not state all the dependency graph.

**Comment by scottgigante-immunai** [comment]
If anyone comes here looking for a solution, I wrote a custom [github actions workflow](https://github.com/openproblems-bio/openproblems/blob/main/.github/workflows/check_r_dependencies.yml) that upgrades packages from a custom requirements file using renv. It's not the most general solution but could be easily adapted to work with a `DESCRIPTION` file instead.

**Comment by dfrankow** [comment]
Note if I were going to implement a feature to warn about out-of-date dependencies, I might try [writing an renv.lock file](https://rstudio.github.io/renv/reference/snapshot.html) and comparing it to the existing one.

**Comment by dfrankow** [comment]
Note https://r-pkgs.org/dependencies.html#other-dependencies says:

> Prior to the roll-out of namespaces in R 2.14.0 in 2011, Depends was the only way to “depend” on another package. Now, despite the name, you should almost always use Imports, not Depends.

**Comment by jscheidtmann** [comment]
It would be cool, if dependabot would provide functionality similar to [oysteR](https://cran.r-project.org/package=oysteR), which searches the sonatype repository for known vulnerabilities in R packages. One would need to see, if this could be adapted to scan only the requirements of a R software package.

**Comment by actuarysailor** [comment]
Yes, RENV for R works a lot like poetry for python.  It would be great if dependabot could maintain RENV.

**Comment by joergsesterhenn** [comment]
You could try the OSV scanner action. It picks up renv.lock and scans for security related issues in dependencies. https://osv.dev/

**Comment by algo7** [comment]
any news on this?
